src/gluonnlp/model/train/language_model.py

```diff
@@ -71,6 +71,9 @@ def __init__(self, mode, vocab_size, embed_size=400, hidden_size=1150, num_layer
         self._drop_e = drop_e
         self._weight_drop = weight_drop
         self._tie_weights = tie_weights
+        self._shared_params = None
+        if 'params' in kwargs:
+            self._shared_params = kwargs['params']
```

Comment:
I think your use-case is not well supported in the current Gluon API and the current solution may rely on some fragile behaviour in ParameterDict? 

The underlying problem is that `params=self.embedding[0].params` which is used in your PR and prior to your PR is not actually well supported currently due to the way that gluon handles parameter prefixes.